src/analyzer/protocol/tcp/TCP.cc

```diff
@@ -1321,6 +1321,14 @@ void TCP_Analyzer::DeliverPacket(int len, const u_char* data, bool is_orig,
 		PacketWithRST();
 		}
 
+	int32 delta_last = update_last_seq(endpoint, seq_one_past_segment, flags, len);
+	endpoint->last_time = current_timestamp;
+
+	int do_close;
+	int gen_event;
+	UpdateStateMachine(current_timestamp, endpoint, peer, base_seq, ack_seq,
+	                   len, delta_last, is_orig, flags, do_close, gen_event);
+
```

Comment:
Did this chunk of code get moved here on purpose?  Don't immediately see what relation this change has with the new gap-tracking logic.

Reply:
Hah yeah I wondered about how to comment on that.  The move is on purpose, and is necessary so that the gap logic can determine whether the gap comes at the end of a connection, where often sequence numbers are off due to various connection closing glitches.  As best as I can figure, moving the state machine update to come earlier shouldn't change anything else ... and the fact that it doesn't lead to any test suite changes is reassuring in that regard.  I could add a comment at the beginning of the block to that effect.